File: src/main/java/com/indiepost/service/RoleService.java
package com.indiepost.service;

import com.indiepost.enums.Types.UserRole;
import com.indiepost.model.Role;

import java.util.List;

/**
 * Created by jake on 8/4/16.
 */
public interface RoleService {
    void save(UserRole role);

    void update(UserRole role);

    void delete(UserRole role);

    Role findById(Long id);

    Role findByName(String name);

    List<Role> findAll();
}

File: src/main/kotlin/com/indiepost/service/RoleService.kt
package com.indiepost.service

import com.indiepost.enums.Types.UserRole
import com.indiepost.model.Role

/**
 * Created by jake on 8/4/16.
 */
interface RoleService {

    fun save(role: UserRole)

    fun update(role: UserRole)

    fun delete(role: UserRole)

    fun findById(id: Long): Role?

    fun findByName(name: String): Role?

    fun findAll(): List<Role>
}
